lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -222,37 +212,78 @@ class InputConfiguration {
 typedef _OnChangeCallback = void Function(EditingState editingState);
 typedef _OnActionCallback = void Function(String inputAction);
 
-/// Wraps the DOM element used to provide text editing capabilities.
+/// Interface defining the template for text editing strategies.
 ///
-/// The backing DOM element could be one of:
+/// The algorithms will be picked in the runtime depending on the concrete
+/// class implementing the interface.
 ///
-/// 1. `<input>`.
-/// 2. `<textarea>`.
-/// 3. `<span contenteditable="true">`.
-class TextEditingElement {
-  /// Creates a non-persistent [TextEditingElement].
+/// These algorithms is expected to differ by operating system and/or browser.
+abstract class TextEditingStrategy {
+  void initializeTextEditing(
+    InputConfiguration inputConfig, {
+    @required _OnChangeCallback onChange,
+    @required _OnActionCallback onAction,
+  });
+
+  /// Place the DOM element to its initial position.
+  ///
+  /// It will be located exactly in the same place with the editable widgets,
+  /// however it's contents and cursor will be invisible.
   ///
-  /// See [TextEditingElement.persistent] to understand what persistent mode is.
-  TextEditingElement(this.owner);
+  /// Users can interact with the element and use the functionalities of the
+  /// right-click menu. Such as copy,paste, cut, select, translate...
+  void initializeElementPosition();
 
-  /// Timer that times when to set the location of the input text.
+  /// Register event listeners to the DOM element.
   ///
-  /// This is only used for iOS. In iOS, virtual keyboard shifts the screen.
-  /// There is no callback to know if the keyboard is up and how much the screen
-  /// has shifted. Therefore instead of listening to the shift and passing this
-  /// information to Flutter Framework, we are trying to stop the shift.
+  /// These event listener will be removed in [disable].
```

Comment:
I think it's important to mention that event listeners need to be added to the `_subscriptions` list in order to be cleaned up properly.